test/metabase/api/notify_test.clj

```diff
@@ -19,17 +19,39 @@
       (is (= (get mw.util/response-forbidden :body)
              (client/client :post 403 "notify/db/100"))))))
 
+(def api-headers {:headers {"X-METABASE-APIKEY" "test-api-key"
+                            "Content-Type"      "application/json"}})
+
 (deftest not-found-test
-  (testing "POST /api/notify/db/:id"
-    (testing "database must exist or we get a 404"
-      (is (= {:status 404
-              :body   "Not found."}
-             (try (http/post (client/build-url (format "notify/db/%d" Integer/MAX_VALUE) {})
-                                    {:accept  :json
-                                     :headers {"X-METABASE-APIKEY" "test-api-key"
-                                               "Content-Type"      "application/json"}})
-                  (catch clojure.lang.ExceptionInfo e
-                    (select-keys (ex-data e) [:status :body]))))))))
+  (mt/with-temporary-setting-values [api-key "test-api-key"]
+    (testing "POST /api/notify/db/:id"
+      (testing "database must exist or we get a 404"
+        (is (= {:status 404
+                :body   "Not found."}
+               (try (http/post (client/build-url (format "notify/db/%d" Integer/MAX_VALUE) {})
+                               (merge {:accept :json} api-headers))
+                    (catch clojure.lang.ExceptionInfo e
+                      (select-keys (ex-data e) [:status :body]))))))
+      (testing "table ID must exist or we get a 404"
+        (is (= {:status 404
+                :body   "Not found."}
+               (try (http/post (client/build-url (format "notify/db/%d" (:id (mt/db))) {})
+                               (merge {:accept       :json
+                                       :content-type :json
+                                       :form-params  {:table_id Integer/MAX_VALUE}}
+                                      api-headers))
+                    (catch clojure.lang.ExceptionInfo e
+                      (select-keys (ex-data e) [:status :body]))))))
+      (testing "table name must exist or we get a 404"
+        (is (= {:status 404
+                :body   "Not found."}
+               (try (http/post (client/build-url (format "notify/db/%d" (:id (mt/db))) {})
+                               (merge {:accept       :json
+                                       :content-type :json
+                                       :form-params  {:table_name "IncorrectToucanFact"}}
+                                      api-headers))
+                    (catch clojure.lang.ExceptionInfo e
+                      (select-keys (ex-data e) [:status :body])))))))))
```

Comment:
Not the DRYest testing, but I don't like to get too cute with function extraction in test.